rfc/0010-cbft.md

```diff
@@ -0,0 +1,1338 @@
+
+## 1. Query Type Overview
+
+There are different types of queries which can be executed, some of them (compound queries) contain other queries in them. They all provide a common set of options and then each one has a unique option set. The following provides an overview of the supported query types by the server. Each of them will be detailed later in the request building section.
+
+Simple Queries:
+
+ - **Match Query:** A match query analyzes the input text and uses that analyzed text to query the index.
+ - **Match Phrase Query:** The input text is analyzed and a phrase query is built with the terms resulting from the analysis.
+ - **Regexp Query:** Finds documents containing terms that match the specified regular expression.
+ - **String Query:** The query string query allows humans to describe complex queries using a simple syntax.
+ - **Wildcard Query:** Interprets * and ? wildcards as found in a lot of applications, for an easy implementation of such a search feature.
+ - **DocId Query:** Allows to restrict matches to a set of specific documents.
+ - **Boolean Field Query:** Allow to match `true`/`false` in a field mapped as boolean (which can reduce index size).
+
+Range Queries:
+
+ - **Date Range Query:** The date range query finds documents containing a date value in the specified field within the specified range.
+ - **Numeric Range Query:** The numeric range query finds documents containing a numeric value in the specified field within the specified range.
+
+Compound Queries:
+
+ - **Conjunction Query:** Result documents must satisfy all of the child queries.
+ - **Disjunction Query:** Result documents must satisfy a configurable min number of child queries.
+ - **Boolean Query:** The boolean query is a useful combination of conjunction and disjunction queries.
+
+Some queries are more limited and usually used for debugging purposes:
+
+ - **TermQuery:** A query that looks for **exact** matches of the term in the index (no analyzer, no stemming). Useful to check what the actual content of the index is. Usual better alternative is `MatchQuery`.
+ - **Fuzzy Query:** A fuzzy query is a term query that matches terms within a specified edit distance (Levenshtein distance). Usual better alternative is `MatchQuery`.
+ - **Prefix Query:** The prefix query finds documents containing terms that start with the provided prefix. Usual better alternative is `MatchQuery`.
+ - **PhraseQuery:** A query that looks for **exact** match of several terms (in the exact order) in the index. Usual better alternative is `MatchPhraseQuery`.
+ - **MatchAllQuery:** A query that matches all indexed documents.
+ - **MatchNoneQuery:** A query that matches nothing.
+
+## 2. High Level API Overview
+This section describes how the overall API looks like (to be expanded).
```

Comment:
We seem to be missing the parameter for the index name to be queried. Seems like it should be defined on the SearchQueryRequest since it represents the endpoint and not the parameters.